libr/cons/dietline.c

```diff
@@ -1948,32 +1967,52 @@ R_API const char *r_line_readline_cb(RLineReadCallback cb, void *user) {
 						break;
 					case '5': // pag up
 						buf[1] = r_cons_readchar ();
+						if (I.hud) {
+							I.hud->top_entry_n += (rows - 1);
+							if (I.hud->top_entry_n + 1 >= I.hud->current_entry_n) {
+								I.hud->top_entry_n = I.hud->current_entry_n;
+							}
+						}
 						if (I.sel_widget) {
 							selection_widget_up (R_MIN (I.sel_widget->h, R_SELWIDGET_MAXH));
 							selection_widget_draw ();
 						}
 						break;
 					case '6': // pag down
 						buf[1] = r_cons_readchar ();
+						if (I.hud) {
+							I.hud->top_entry_n -= (rows - 1);
+							if (I.hud->top_entry_n < 0) {
+								I.hud->top_entry_n = 0;
+							}
+						}
 						if (I.sel_widget) {
 							selection_widget_down (R_MIN (I.sel_widget->h, R_SELWIDGET_MAXH));
 							selection_widget_draw ();
 						}
 						break;
 					/* arrows */
 					case 'A':	// up arrow
-						if (I.sel_widget) {
+						if (I.hud) {
+							if (I.hud->top_entry_n + 1 < I.hud->current_entry_n) {
+								I.hud->top_entry_n++;
+							}
+						} else if (I.sel_widget) {
 							selection_widget_up (1);
 							selection_widget_draw ();
 						} else if (gcomp) {
 							gcomp_idx++;
 						} else if (r_line_hist_up () == -1) {
 							r_cons_break_pop ();
 							return NULL;
-						}
+						} 
```

Comment:
You forgot trailing whitespace though ;)

Reply:
My bad, fixed it :)